### DISCUSSION TITLE
Using batch corrected scRNA-seq data for CCA integration with scATAC-seq data

### DISCUSSION THREAD

**Discussion by RegnerM2015** [stuart-lab/signac#587] [Q&A] [answered]
Hi @timoast,

Is it generally advisable to use batch corrected scRNA-seq data for the RNA/ATAC label transfer procedure?

For example, if there are two pairs of matching scRNA-seq/scATAC-seq datasets collected on different dates, would you run CCA to integrate the two scRNA-seq datasets first? Then use this integrated scRNA-seq dataset for the CCA label transfer to scATAC-seq? This seems unreasonable/dangerous to use CCA twice. 

Are there other batch correction methods that would work in this scenario? For example, instead of using CCA to integrate the two scRNA-seq datasets, could you use Harmony instead? Then run the CCA label transfer procedure using the Harmony corrected scRNA-seq dataset? 

This recent paper batch corrected their scRNA-seq data with Harmony, before label transferring to their scATAC-seq data: https://www.nature.com/articles/s41467-021-22368-w#Sec12

Is this recommended?

Thanks!

**Comment by timoast** [comment, marked as answer]
You can certainly run multimodal label transfer between batch-corrected scRNA-seq and scATAC-seq. 

> This seems unreasonable/dangerous to use CCA twice.

In this case you are not performing CCA twice per se, you're first running CCA and finding anchors between scRNA-seq datasets, then using those anchors to correct the gene expression vectors for each cell. You're then running CCA on the corrected gene expression and the scATAC-seq gene activity matrix.

> This recent paper batch corrected their scRNA-seq data with Harmony, before label transferring to their scATAC-seq data

Harmony corrects the low-dimensional cell embeddings, and so would not influence mutlimodal integration (which uses the gene expression data)

**Reply by RegnerM2015** [reply]
Thank you so much! This answers my question.

**Comment by RegnerM2015** [comment]
Continuing our conversation @timoast,

Considering the following hypothetical scenario:

The goal is to transfer labels between matched scRNA-seq and scATAC-seq datasets for a total of 10 tissue samples. 

1) We combine the 10 scRNA-seq datasets using CCA integration to create one multi-sample scRNA-seq dataset and recluster the dataset with Louvain clustering. 

2) We combine the 10 scATAC-seq datasets using Harmony integration to create one multi-sample scATAC-seq dataset. 

3) Finally, we use CCA to transfer labels from the multi-sample scRNA-seq dataset to the multi-sample scATAC-seq dataset. 

Are these steps and strategies reasonable? Is it okay to use CCA batch correction in scRNA-seq but Harmony integration in scATAC-seq?
Or would it be preferable to use Harmony integration for both scRNA-seq and scATAC-seq before transfering labels? 

Thanks for your help!

**Reply by timoast** [reply]
This sounds fine but I just want to point out again that the Harmony integration is not used at all during the scRNA/scATAC-seq integration, since Harmony corrects the low-dimensional cell embeddings. In the scRNA/scATAC integration you're using the "corrected" scRNA gene expression, and the "uncorrected" scATAC gene activities. Basically, you could omit the Harmony step and get identical results here.

**Reply by RegnerM2015** [reply]
@timoast 

I understand the ATAC gene activities would not change, only the low dimensional embeddings. However, I thought the low dimensional embedding space **did** matter when you transfer labels.

I'm referring to this piece of code from the tutorial at https://satijalab.org/seurat/articles/atacseq_integration_vignette.html:
```
celltype.predictions <- TransferData(anchorset = transfer.anchors, refdata = pbmc.rna$seurat_annotations,
    weight.reduction = pbmc.atac[["lsi"]], dims = 2:30)

pbmc.atac <- AddMetaData(pbmc.atac, metadata = celltype.predictions)
```

If we used pbmc.atac[["Harmony-LSI"]] instead of pbmc.atac[["lsi"]] for the weight reduction parameter, would the results change?

**Reply by timoast** [reply]
Yes that's a good point, if you're setting `weight.reduction` to the Harmony-corrected LSI then it's being used for the anchor weighting

**Reply by RegnerM2015** [reply]
Correct, I mis-asked my question then. 

So Harmony correction does influence anchor weighting, and may lead to different results when transferring labels from RNA to ATAC?

**Reply by timoast** [reply]
Right

**Reply by RegnerM2015** [reply]
Sounds good! Thank you for your help again!

**Comment by zrcjessica** [comment]
Hi, I wanted to follow up on this question - can I use the same approach to do label transfer from a set of snRNA-seq data that was integrated using RPCA instead of CCA?

**Reply by timoast** [reply]
Yes, I don't think it matters how the scRNA-seq was integrated
